include/exec/linux/__detail/io_uring_context.hpp

```diff
@@ -234,6 +223,17 @@ namespace exec { namespace __io_uring {
     , __completion_queue_{__completion_queue_region_ ? __completion_queue_region_ : __submission_queue_region_, __params_}
     , __submission_queue_{__submission_queue_region_, __submission_queue_entries_, __params_}
     , __wakeup_operation_{this, __eventfd_} {
+    __wakeup_operation_.start();
+  }
+
+  inline __context::~__context() {
+    // It is a fatal error if the context is destroyed while it is still in a running state.
+    STDEXEC_ASSERT(!__is_running_.load(std::memory_order_relaxed));
+    if (__n_submissions_in_flight_.load(std::memory_order_relaxed) != __no_new_submissions) {
+      request_stop();
+      __break_loop_.store(false, std::memory_order_release);
+      run();
+    }
```

Comment:
I know @kirkshoop has strong feelings about blocking in destructors. Kirk, do you want to chime in here?

Reply:
> I know @kirkshoop has strong feelings about blocking in destructors. Kirk, do you want to chime in here?

Yes. You are right, blocking within the destructor is bad and this is a consequence of unconditionally breaking from the run loop. I could change the semantics to "stop executing the run loop if no I/O operations are ongoing instead of waiting for more ops to arrive".

Reply:
Done